Jenkinsfile_k8s

```diff
@@ -100,13 +100,44 @@ pipeline {
             }
           }
           stage('Helmfile Lint'){
+            // Always lint on main branch
+            // Lint on pull requests only if the current cluster helmfile or any of the chart configurations files have changed
+            when {
+              anyOf {
+                branch 'main'
+                allOf {
+                  changeRequest()
+                  anyOf {
+                    changeset "config/**"
+                    expression {
+                      sh(returnStatus: true, script: '''
+                        git status -s | grep "^clusters/${K8S_CLUSTER}.yaml"
+                        '''
+                      ) == 0
+                    }
+                  }
+                }
+              }
+            }
             steps {
               sh 'helmfile -f "clusters/${K8S_CLUSTER}.yaml" lint'
             }
           } // stage
           stage('Diff on Pull Request'){
+            // Diff on pull requests only if the current cluster helmfile or any of the chart configurations files have changed
             when {
-              changeRequest()
+              allOf {
+                changeRequest()
+                anyOf {
+                  changeset "config/**"
+                  expression {
+                    sh(returnStatus: true, script: '''
+                      git status -s | grep "^clusters/${K8S_CLUSTER}.yaml"
```

Comment:
doesn't changeset let you do that?

Reply:
`when { changeset pattern: "^clusters/${K8S_CLUSTER}.yaml", comparator: "REGEXP" }`

Reply:
I've tried before with `changeset "clusters/${K8S_CLUSTER}.yaml" but it didn't worked. (Tested again with a replay here: https://infra.ci.jenkins.io/job/kubernetes-jobs/job/kubernetes-management/job/PR-2852/3/console)

I've just tried your syntax in another replay but it doesn't seem to work either: https://infra.ci.jenkins.io/job/kubernetes-jobs/job/kubernetes-management/job/PR-2852/4/console

Unfortunately, git+grep doesn't seem to work too: https://infra.ci.jenkins.io/job/kubernetes-jobs/job/kubernetes-management/job/PR-2852/5/console

Reply:
well grep by default isn't using regex, so you want egrep or grep -e

I'll try to take a look at those links when i'm on a computer i can run vpn again.